### DISCUSSION TITLE
Bubble-card type Separator, the sub-button  style does not recognise states.

### DISCUSSION THREAD

**Discussion by J-PD** [Clooos/Bubble-Card#1026] [Questions about config, custom styles and templates] [answered]
Hi,

I'm rebuilding my dashboard using Bubble cards and was able to overcome all my styling challenges except for this one, I hoop someone can help me out.

I created a Bubble card  with type  Separator and added a sub-button to turn on/off a media player. I added following to the Custom styles/templates configuration.

`type: custom:bubble-card
card_type: separator
name: Tuin
icon: mdi:music
styles: |
  .bubble-line {
    opacity: 0.8 !important;
    height: 4px;
    background-color: white !important;
    margin-right: 14px;
  }
  .bubble-sub-button-icon {
    --mdc-icon-size: 30px !important;
    color: ${state === 'on' ? 'rgb(0, 250, 0)' : 'rgb(250, 0, 0)' } !important ; 
  }
  .bubble-sub-button {
    background-color: ${state === 'on' ? 'rgb(0, 70, 0)' : 'rgb(70, 0, 0)' } !important ;
  }
card_layout: large
sub_button:
  - entity: switch.sh150_y002
    icon: mdi:power
    show_state: true
    tap_action:
      action: toggle
`

Everything works as I want except that the color of the sub-button (.bubble-sub-button and .bubble-sub-button-icon) does not change with the state. I have no idea what I'm doing wrong at this point and was hoping that someone can point me in the good direction.

This is how it looks on my dashboard when the media player is power off,
![Separator_1](https://github.com/user-attachments/assets/4979f86e-d1ca-4ddf-94a2-16fe442f343d)

and when the media player is power on I get this where the sub-button should be green in stead of red.
![Separator_2](https://github.com/user-attachments/assets/b9cc80f1-0150-409a-a443-9d805746df1b)

**Comment by MrBearPresident** [comment, marked as answer]
'state' is linked to the main entity of the bubble-card.
You have to use `hass.states['entityID'].state` instead.

---

<sup> If this solves your issue please mark it as solved and close it. </sup>

**Reply by J-PD** [reply]
Thank you, that was indeed the solution, 
![image](https://github.com/user-attachments/assets/65b1e1cf-605a-40c6-84cd-cf29bec83a63)

Thanks again for this super fast answer .
